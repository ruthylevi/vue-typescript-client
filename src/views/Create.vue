<template>
  <div>
    <h1>This is a create page</h1>
  </div>
</template>
