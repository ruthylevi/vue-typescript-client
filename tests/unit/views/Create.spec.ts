import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import { shallowMount } from '@vue/test-utils';
import Create from '@/views/Create.vue';

Vue.use(Vuetify);
Vue.use(VueRouter);

const wrapper = shallowMount(Create, { stubs: ['PictureInput'] });

describe('Create', () => {
  test('Sanity', () => {
    expect(true).toBe(true);
  });

  test('Visible', () => {
    expect(wrapper.isVisible()).toBe(true);
  });

  test('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
