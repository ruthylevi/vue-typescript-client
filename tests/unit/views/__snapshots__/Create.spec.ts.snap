// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Renders correctly 1`] = `
<v-container-stub tag="div">
  <v-row-stub tag="div">
    <v-col-stub cols="12" sm="8" tag="div">
      <h1>Create Project</h1> <br>
      <div>
        <v-card-stub loaderheight="4" tag="div">
          <v-container-stub tag="div">
            <v-row-stub tag="div">
              <v-col-stub cols="12" sm="3" tag="div">
                <v-file-input-stub errorcount="1" errormessages="" messages="" rules="value => !value || value.size < 2000000 || 'Avatar size should be less than 2 MB!'" successmessages="" backgroundcolor="" prependicon="mdi-camera" loaderheight="2" clearable="true" clearicon="$clear" placeholder="Avatar" type="file" countersizestring="$vuetify.fileInput.counterSize" counterstring="$vuetify.fileInput.counter" truncatelength="22" accept="image/png, image/jpeg, image/bmp"></v-file-input-stub>
              </v-col-stub>
              <v-col-stub cols="12" sm="9" tag="div">
                <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Project Title" loaderheight="2" clearicon="$clear" type="text"></v-text-field-stub>
              </v-col-stub>
            </v-row-stub>
            <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Project Card Description" loaderheight="2" clearicon="$clear" type="text"></v-text-field-stub>
          </v-container-stub>
        </v-card-stub>
        <p></p>
        <v-card-stub loaderheight="4" tag="div">
          <v-container-stub tag="div"><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <pictureinput-stub width="600" height="600" margin="16" accept="image/jpeg, image/png" size="10" buttonclass="btn btn-primary button" removebuttonclass="v-btn" aspectbuttonclass="btn btn-secondary button secondary" prefill="" prefilloptions="[object Object]" crop="true" radius="0" removable="true" hidechangebutton="true" changeonclick="true" zindex="10000" alertonerror="true" customstrings="[object Object]"></pictureinput-stub> <br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
          </v-container-stub>
        </v-card-stub>
        <p></p>
      </div>
      <p></p>
    </v-col-stub>
    <v-col-stub cols="12" sm="4" tag="div">
      <v-card-stub loaderheight="4" tag="div">
        <v-btn-stub tag="button" activeclass="" block="true" outlined="true" type="button">
          <h2>Submit</h2>
        </v-btn-stub>
        <v-dialog-stub opendelay="0" closedelay="0" contentclass="" hideoverlay="true" maxwidth="none" origin="center center" persistent="true" retainfocus="true" transition="dialog-transition" width="300">
          <v-card-stub loaderheight="4" tag="div" color="primary" dark="true">
            <v-card-text-stub>
              Please stand by
              <v-progress-linear-stub color="white" value="0" active="true" buffervalue="100" height="4" indeterminate="true" class="mb-0"></v-progress-linear-stub>
            </v-card-text-stub>
          </v-card-stub>
        </v-dialog-stub>
      </v-card-stub>
      <p></p>
      <v-card-stub loaderheight="4" tag="div">
        <v-col-stub cols="12" sm="6" md="12" tag="div">
          <h2>Add Project Info:</h2> <br>
          <v-textarea-stub errorcount="1" errormessages="" messages="" rules="v => (v || '').length <= len || \`Invalid character length, must be less than \${len}\`" successmessages="" backgroundcolor="" label="Description" loaderheight="2" clearicon="$clear" counter="2048" outlined="true" type="text" rowheight="24" rows="5" name="description"></v-textarea-stub>
          <v-row-stub tag="div" align="center">
            <v-col-stub cols="12" tag="div">
              <v-autocomplete-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" appendicon="$dropdown" backgroundcolor="" dense="true" label="Majors" loaderheight="2" clearicon="$clear" outlined="true" type="text" valuecomparator="function deepEqual(a, b) {
          if (a === b) return true;

          if (a instanceof Date &amp;&amp; b instanceof Date) {
            // If the values are Date, they were convert to timestamp with getTime and compare it
            if (a.getTime() !== b.getTime()) return false;
          }

          if (a !== Object(a) || b !== Object(b)) {
            // If the values aren't objects, they were already checked for equality
            return false;
          }

          var props = Object.keys(a);

          if (props.length !== Object.keys(b).length) {
            // Different number of props, don't bother to check
            return false;
          }

          return props.every(function (p) {
            return deepEqual(a[p], b[p]);
          });
        }" nodatatext="$vuetify.noDataText" chips="true" items="Software Engineering,Computer Science,Computer Engineering,Mechanical Engineering,Engineering Management,Electrical Engineering,Business and Technology,Civil Engineering,Environmental Engineering,Chemical Engineering,Biomedical Engineering,Music &amp; Technology,Physics,Naval Engineering,Pure and Applied Mathematics,Engineering Physics,Cybersecurity,Chemistry,Biology,Chemical Biology,Computational Science,Industrial and Systems Engineering,Accounting &amp; Analytics,Finance,Quantitative Finance,Marketing Innovation &amp; Analytics,Information Systems,Economics,Management,History,Philosophy,Social Sciences,Literature,Science Communication,Visual Arts &amp; Technology,Science, Technology, and Society" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" multiple="true" smallchips="true" allowoverflow="true" filter="function _default(item, queryText, itemText) {
                return itemText.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1;
              }"></v-autocomplete-stub>
              <v-autocomplete-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" appendicon="$dropdown" backgroundcolor="" dense="true" label="Tags" loaderheight="2" clearicon="$clear" outlined="true" type="text" valuecomparator="function deepEqual(a, b) {
          if (a === b) return true;

          if (a instanceof Date &amp;&amp; b instanceof Date) {
            // If the values are Date, they were convert to timestamp with getTime and compare it
            if (a.getTime() !== b.getTime()) return false;
          }

          if (a !== Object(a) || b !== Object(b)) {
            // If the values aren't objects, they were already checked for equality
            return false;
          }

          var props = Object.keys(a);

          if (props.length !== Object.keys(b).length) {
            // Different number of props, don't bother to check
            return false;
          }

          return props.every(function (p) {
            return deepEqual(a[p], b[p]);
          });
        }" nodatatext="$vuetify.noDataText" chips="true" items="Programming,Artificial Intelligence,Machine Learning,Deep Learning,Data Mining,Virtual Reality,Augmented Reality,Automotive,Robotics,Mechatronics /Automation,Thermal / fluids,Aviation / Aerospace Systems,Biomedical,Product Development,Energy / Sustainability,Competition,Thermal / Fluids / Energy Sustainability,Site / Civil Design,Structural,Transportation" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" multiple="true" smallchips="true" allowoverflow="true" filter="function _default(item, queryText, itemText) {
                return itemText.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1;
              }" autocomplete="off"></v-autocomplete-stub>
            </v-col-stub>
          </v-row-stub>
        </v-col-stub>
      </v-card-stub> <br>
      <v-card-stub loaderheight="4" tag="div">
        <v-container-stub tag="div">
          <h2>Add Members:</h2> <br>
          <v-autocomplete-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" appendicon="$dropdown" backgroundcolor="" dense="true" label="Advisors" loaderheight="2" clearicon="$clear" outlined="true" type="text" valuecomparator="function deepEqual(a, b) {
          if (a === b) return true;

          if (a instanceof Date &amp;&amp; b instanceof Date) {
            // If the values are Date, they were convert to timestamp with getTime and compare it
            if (a.getTime() !== b.getTime()) return false;
          }

          if (a !== Object(a) || b !== Object(b)) {
            // If the values aren't objects, they were already checked for equality
            return false;
          }

          var props = Object.keys(a);

          if (props.length !== Object.keys(b).length) {
            // Different number of props, don't bother to check
            return false;
          }

          return props.every(function (p) {
            return deepEqual(a[p], b[p]);
          });
        }" nodatatext="$vuetify.noDataText" chips="true" items="Gregg Vesonder,Leslie Brunell,Alex De Rosa,Long Wang,Brendan Englot" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" multiple="true" smallchips="true" allowoverflow="true" filter="function _default(item, queryText, itemText) {
                return itemText.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1;
              }"></v-autocomplete-stub>
          <v-autocomplete-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" appendicon="$dropdown" backgroundcolor="" dense="true" label="Students" loaderheight="2" clearicon="$clear" outlined="true" type="text" valuecomparator="function deepEqual(a, b) {
          if (a === b) return true;

          if (a instanceof Date &amp;&amp; b instanceof Date) {
            // If the values are Date, they were convert to timestamp with getTime and compare it
            if (a.getTime() !== b.getTime()) return false;
          }

          if (a !== Object(a) || b !== Object(b)) {
            // If the values aren't objects, they were already checked for equality
            return false;
          }

          var props = Object.keys(a);

          if (props.length !== Object.keys(b).length) {
            // Different number of props, don't bother to check
            return false;
          }

          return props.every(function (p) {
            return deepEqual(a[p], b[p]);
          });
        }" nodatatext="$vuetify.noDataText" chips="true" items="Ruthy Levi,Herbie Zieger,Mark Freeman,Jason Chlus,Julia Cahn,Tyler Lowe,Jakub Kolasinski" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" multiple="true" smallchips="true" allowoverflow="true" filter="function _default(item, queryText, itemText) {
                return itemText.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1;
              }"></v-autocomplete-stub>
        </v-container-stub>
      </v-card-stub>
      <p></p>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
